nix/README.md

```diff
@@ -0,0 +1,77 @@
+# Nix development and build environment
+
+With Nix it's possible to quickly and reliably recreate the full environments
+for developing, testing and building PostgREST.
+
+## Getting started with Nix
+
+You'll need to [get Nix](https://nixos.org/download.html). The installer will
+create your Nix store in the `/nix/` directory, where all build artifacts and
+their dependencies will be stored. It will also link the Nix executables like
+`nix-env`, `nix-build` and `nix-shell` into your PATH. Nix will manage all
+other PostgREST dependencies from here on out. To clean up older build
+artifacts from the `/nix/store`, you can run `nix-collect-garbage`.
+
+## Building PostgREST
+
+To build PostgREST from your local checkout of the repository, run:
+
+```bash
+nix-build --attr postgrest
+
+```
+
+This will create a `result` directory that contains the PostgREST binary at
+`result/bin/postgrest`. The `--attr` parameter (or short: `-A`) tells Nix to
+build the `postgrest` attribute from the Nix expression it finds in our
+`default.nix` (see below for details). Nix will take care of getting the right
+GHC version and all the build dependencies.
+
+## Developing
+
+A development environment for PostgREST is available with `nix-shell`. The
+following command will put you into a new shell that has GHC and Cabal on the
+PATH:
+
+```bash
+nix-shell
+
+```
+
+Within `nix-shell`, you can run Cabal commands as usual.
```

Comment:
Is only cabal meant to work? I'm trying building with stack(inside nix-shell) and I get:
```
stack build --nix
error: attribute 'ghc883' missing, at (string):1:74
```

Reply:
For now yes - the stack `--nix` integration is a bit more tricky to get right, Cabal seems to be the recommended way as far as I could see. Stack still works as before, both within and without nix-shell, if we don't use '--nix'. You can also try adding `pkgs.stack` to `shell.nix` to get the updated/compatible version.

Reply:
Oh, I see. I'm fine with using cabal inside nix-shell. However, that would mean these [ghc-options defined in stack.yaml](https://github.com/PostgREST/postgrest/blob/master/stack.yaml#L3-L10) won't be used by cabal right? Shouldn't those be added?

Reply:
Yes, that's true! I'll look into adding those.

Reply:
The cleanest way is to add the flags to the .cabal file directly, I think. See https://github.com/PostgREST/postgrest/pull/1489/commits/6f461f4c522370eaf259a00042aaaf9451691f5b for my suggestion.

We could also override the flags directly through Nix. For example, in `default.nix` we'd need to add something like:

```nix
  postgrest =
    pkgs.lib.overrideDerivation drv (
      base: {
        configureFlags =
          base.configureFlags ++ [
            "--ghc-option=-O2"
            "--ghc-option=-Werror"
            "--ghc-option=-Wall"
            ...
          ];
      }
    );
```

(where `overrideDerivation` is a function that takes a derivation and a function that returns a modified derivation)

This seems less maintainable and has the downside that other tools that process Hackage/Stackage would not pick the flags up. E.g. Nixpkgs only looks at the .cabal file of each package and essentially runs 'cabal2nix' on it.